training/src/app/chuck-noris/search/search.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { debounceTime, distinctUntilChanged, Subject, takeUntil, tap } from "rxjs";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+	public searchForm: FormGroup;
+	@Output() public generateJokesFromSearch: EventEmitter<string> = new EventEmitter<string>();
+	@Output() public generateJokesFromClick: EventEmitter<string>  = new EventEmitter<string>();
+	public shouldDestroy: Subject<void>                            = new Subject<void>();
+
+	constructor(private fb: FormBuilder) {
+	}
+
+	ngOnInit(): void {
+		this.searchForm = this.fb.group({
+			searchInput: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(120)]]
+		});
+
+		this.searchForm.get('searchInput')?.valueChanges
+		.pipe(
+			tap((term: string) => {
+				if (term != this.searchForm.get('searchInput').value) {
+					if (this.searchForm.get('searchInput')?.valid) {
```

Comment:
can cause a bug.
lets say the user searched for "hello" and got jokes.
now he search for invalid value "he", and probably the request will go with old value of "hello"
try it